syntax="proto3";
package jarvis;

import "table.proto";

option cc_generic_services = true;

message HttpRequest { };
message HttpResponse { };

message financial_request {
    financial record = 1;
};
message financial_response {
    enum rcode {
        ok = 0;
        dberr = 1;
    }
    rcode code     =  1;
    repeated financial records = 2;
};


service Jarvis {
    rpc TestQuery(HttpRequest) returns (HttpResponse);
    rpc AppendFinancialRecord(financial_request) returns (financial_response);
    rpc DeleteFinancialRecord(financial_request) returns (financial_response);
}